tests/lib/libaa-query-peek/query.c

```diff
@@ -0,0 +1,65 @@
+#include <dlfcn.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/apparmor.h>
+
+static void *libaa_handle = NULL;
+static int (*real_aa_query_label)(uint32_t mask, char *query, size_t size, int *allowed, int *audited);
+
+__attribute__((constructor)) static void init() {
+    char *error;
+
+    libaa_handle = dlopen("libapparmor.so.1", RTLD_LAZY);
+    if (!libaa_handle) {
+        fprintf(stderr, "cannot open libapparmor.so.1: %s\n", dlerror());
+        exit(EXIT_FAILURE);
+    }
+
+    (void)dlerror(); /* Clear any existing error */
```

Comment:
hm I don't recall dlerror() clearing any errors. I suppose this was copied from the man page example? In any case, I'm not sure what the call is supposed to clear if the error path was already handled in the `if()` branch just above.